crates/hyperswitch_domain_models/src/router_data.rs

```diff
@@ -1179,7 +1175,7 @@ impl
         &self,
         payment_data: &payments::PaymentStatusData<router_flow_types::PSync>,
     ) -> Option<MinorUnit> {
-        let payment_attempt = payment_data.payment_attempt.as_ref()?;
+        let payment_attempt = payment_data.payment_attempt.clone();
```

Comment:
```suggestion
        let payment_attempt = &payment_data.payment_attempt;
```
Try to avoid cloning. Always work with references.

Reply:
sure